### PR TITLE
PrivGetLogs needs to check privacy group membership at given block

### PR DESCRIPTION
**Pull request by macfarla**
Signed-off-by: Sally MacFarlane <sally.macfarlane@consensys.net>

When getting private logs, pass block number through so group membership can be checked at that point in time.

See #882

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetLogs.java [hunk 1]
```diff
@@ -85,9 +100,17 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
         requestContext.getRequest().getId(), new LogsResult(matchingLogs));
   }
 
+  private List<LogWithMetadata> getEmptyList() {
+    return Collections.emptyList();
+  }
+
   private void checkIfPrivacyGroupMatchesAuthenticatedEnclaveKey(
```
Comment (lucassaldanha) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetLogs.java:107]: We could name this method something like `checkIfAuthenticatedUserWasMemberAtBlock`. This way we reduce the cognitive gap. Not a big deal though.

Or if you like `checkIfPrivacyGroupMatchesAuthenticatedEnclaveKey ` keep it!

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetLogs.java [hunk 2]
```diff
@@ -70,13 +71,27 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
         filter
             .getBlockHash()
             .map(
-                blockHash ->
-                    privacyQueries.matchingLogs(privacyGroupId, blockHash, filter.getLogsQuery()))
+                blockHash -> {
```
Comment (lucassaldanha) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetLogs.java:74]: We could encapsulate the `map` and `orElseGet` blocks into their own functions to make this main function a bit cleaner. Something like `findLogsForBlockHash` and `findLogsForBlockRange`

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetLogs.java [hunk 3]
```diff
@@ -85,9 +100,17 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
         requestContext.getRequest().getId(), new LogsResult(matchingLogs));
   }
 
+  private List<LogWithMetadata> getEmptyList() {
```
Comment (lucassaldanha) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetLogs.java:103]: Do we need this extra method? Why not return `Collections.emptyList()` straight away on line 78?

Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetLogs.java:103]: compile error!

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetLogs.java (hunk 1 above)
Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetLogs.java:107]: memberAtBlock makes more sense.

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetLogs.java (hunk 2 above)
Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetLogs.java:74]: done

**Pull request merged**